pkg/controller/controller.go

```diff
@@ -68,6 +70,7 @@ func NewController(controllerConfig *spec.ControllerConfig) *Controller {
 		config:           *controllerConfig,
 		opConfig:         &config.Config{},
 		logger:           logger.WithField("pkg", "controller"),
+		controllerID:     os.Getenv("CONTROLLER_ID"),
```

Comment:
Because of the way [`os.Getenv` works](https://golang.org/pkg/os/#Getenv), setting `CONTROLLER_ID` to the empty string (e.g. due to a mistake in a CI/CD pipeline) will have the same effect as not setting it at all. That might lead to  a hard-to-spot misconfiguration with undefined consequences. using [LookupEnv](https://golang.org/pkg/os/#LookupEnv) with a warning for such a case may be slightly better